In [3]:
import pandas as pd

In [16]:
bank_data = pd.read_csv("bank.csv", sep=";")

In [17]:
bank_data 

age            job   marital  education default  balance housing loan  \
0      30     unemployed   married    primary      no     1787      no   no   
1      33       services   married  secondary      no     4789     yes  yes   
2      35     management    single   tertiary      no     1350     yes   no   
3      30     management   married   tertiary      no     1476     yes  yes   
4      59    blue-collar   married  secondary      no        0     yes   no   
5      35     management    single   tertiary      no      747      no   no   
6      36  self-employed   married   tertiary      no      307     yes   no   
7      39     technician   married  secondary      no      147     yes   no   
8      41   entrepreneur   married   tertiary      no      221     yes   no   
9      43       services   married    primary      no      -88     yes  yes   
10     39       services   married  secondary      no     9374     yes   no   
11     43         admin.   married  secondary      no      264     yes   no   
12     36     technician   married   tertiary      no     1109      no   no   
13     20        student    single  secondary      no      502      no   no   
14     31    blue-collar   married  secondary      no      360     yes  yes   
15     40     management   married   tertiary      no      194      no  yes   
16     56     technician   married  secondary      no     4073      no   no   
17     37         admin.    single   tertiary      no     2317     yes   no   
18     25    blue-collar    single    primary      no     -221     yes   no   
19     31       services   married  secondary      no      132      no   no   
20     38     management  divorced    unknown      no        0     yes   no   
21     42     management  divorced   tertiary      no       16      no   no   
22     44       services    single  secondary      no      106      no   no   
23     44   entrepreneur   married  secondary      no       93      no   no   
24     26      housemaid   married   tertiary      no      543      no   no   
25     41     management   married   tertiary      no     5883      no   no   
26     55    blue-collar   married    primary      no      627     yes   no   
27     67        retired   married    unknown      no      696      no   no   
28     56  self-employed   married  secondary      no      784      no  yes   
29     53         admin.   married  secondary      no      105      no  yes   
...   ...            ...       ...        ...     ...      ...     ...  ...   
4491   35    blue-collar    single  secondary      no        0     yes   no   
4492   32     technician    single  secondary      no      309     yes  yes   
4493   28     technician    single   tertiary      no        0     yes   no   
4494   26     technician    single  secondary      no      668     yes   no   
4495   48     management   married   tertiary      no     1175     yes   no   
4496   30    blue-collar    single  secondary      no      363      no   no   
4497   31   entrepreneur    single   tertiary      no       38      no   no   
4498   31     management   married   tertiary      no     1183     yes   no   
4499   45    blue-collar  divorced    primary      no      942      no   no   
4500   38         admin.   married  secondary      no     4196     yes   no   
4501   34     management   married   tertiary      no      297     yes   no   
4502   42       services   married  secondary      no      -91     yes  yes   
4503   60  self-employed   married    primary      no      362      no  yes   
4504   42    blue-collar    single  secondary      no     1080     yes  yes   
4505   32         admin.    single  secondary      no      620     yes   no   
4506   42     unemployed  divorced   tertiary      no     -166      no   no   
4507   33       services   married  secondary      no      288     yes   no   
4508   42         admin.   married    unknown      no      642     yes  yes   
4509   51     technician   married   tertiary      no   

In [18]:
bank_data.head()

age          job  marital  education default  balance housing loan  \
0   30   unemployed  married    primary      no     1787      no   no   
1   33     services  married  secondary      no     4789     yes  yes   
2   35   management   single   tertiary      no     1350     yes   no   
3   30   management  married   tertiary      no     1476     yes  yes   
4   59  blue-collar  married  secondary      no        0     yes   no   

    contact  day month  duration  campaign  pdays  previous poutcome   y  
0  cellular   19   oct        79         1     -1         0  unknown  no  
1  cellular   11   may       220         1    339         4  failure  no  
2  cellular   16   apr       185         1    330         1  failure  no  
3   unknown    3   jun       199         4     -1         0  unknown  no  
4   unknown    5   may       226         1     -1         0  unknown  no

So, the data.head function returns to us, the first n variables about clients in bank. In this table,we are able to set how many 'n' variables can be shown in table. Main benefit of using this function is that it shows the right using of columns and data types. Because of this function has no suggestions for this one, they automatically give informations about first 5 clients.

In [19]:
bank_data.tail()

age            job  marital  education default  balance housing loan  \
4516   33       services  married  secondary      no     -333     yes   no   
4517   57  self-employed  married   tertiary     yes    -3313     yes  yes   
4518   57     technician  married  secondary      no      295      no   no   
4519   28    blue-collar  married  secondary      no     1137      no   no   
4520   44   entrepreneur   single   tertiary      no     1136     yes  yes   

       contact  day month  duration  campaign  pdays  previous poutcome   y  
4516  cellular   30   jul       329         5     -1         0  unknown  no  
4517   unknown    9   may       153         1     -1         0  unknown  no  
4518  cellular   19   aug       151        11     -1         0  unknown  no  
4519  cellular    6   feb       129         4    211         3    other  no  
4520  cellular    3   apr       345         2    249         7    other  no

Vice versa function for previous one is data.tail. It's show to us the last variables about clients in the table. You finished your work, and if you wanna check - use this function.

In [20]:
bank_data.tail(2)

age           job  marital  education default  balance housing loan  \
4519   28   blue-collar  married  secondary      no     1137      no   no   
4520   44  entrepreneur   single   tertiary      no     1136     yes  yes   

       contact  day month  duration  campaign  pdays  previous poutcome   y  
4519  cellular    6   feb       129         4    211         3    other  no  
4520  cellular    3   apr       345         2    249         7    other  no

We see that in column variable is 2. It means to show your last 2 clients information to us.

In [21]:
bank_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4521 entries, 0 to 4520
Data columns (total 17 columns):
age          4521 non-null int64
job          4521 non-null object
marital      4521 non-null object
education    4521 non-null object
default      4521 non-null object
balance      4521 non-null int64
housing      4521 non-null object
loan         4521 non-null object
contact      4521 non-null object
day          4521 non-null int64
month        4521 non-null object
duration     4521 non-null int64
campaign     4521 non-null int64
pdays        4521 non-null int64
previous     4521 non-null int64
poutcome     4521 non-null object
y            4521 non-null object
dtypes: int64(7), object(10)
memory usage: 600.5+ KB


Main important items of all functions and variables are available in this data.info function. You set up settings to functions,and wants to share with others, you can use this one.  

In [26]:
#correlation 

In [28]:
bank_data.describe()

age       balance          day     duration     campaign  \
count  4521.000000   4521.000000  4521.000000  4521.000000  4521.000000   
mean     41.170095   1422.657819    15.915284   263.961292     2.793630   
std      10.576211   3009.638142     8.247667   259.856633     3.109807   
min      19.000000  -3313.000000     1.000000     4.000000     1.000000   
25%      33.000000     69.000000     9.000000   104.000000     1.000000   
50%      39.000000    444.000000    16.000000   185.000000     2.000000   
75%      49.000000   1480.000000    21.000000   329.000000     3.000000   
max      87.000000  71188.000000    31.000000  3025.000000    50.000000   

             pdays     previous  
count  4521.000000  4521.000000  
mean     39.766645     0.542579  
std     100.121124     1.693562  
min      -1.000000     0.000000  
25%      -1.000000     0.000000  
50%      -1.000000     0.000000  
75%      -1.000000     0.000000  
max     871.000000    25.000000

Summary of statistic and numerical part given in this function. As we see, it has several concluds to each of them. And in the first column, we notice that count of all variables equal to 4521.00000. It gives us sum of all input variables which relates to the numerical way.Important of its count,just shows the sum of number variables. The next is mean, it tells us statistic average of all numbers(data types). We have informations of 4520 people, and we took their age,balance,day,etc and by calculations we took these numbers. Even mean shows us the total average, we use std to be closer to average number of variables(in detaily). If your variables are larger, consequently your std also will be higher. That's why std of balance is the highest than other std variables.Other functions and numbers like 25,50,75% and min,max functions gives us the total sum of all variables which shown in table about customers. By math way,they calculate all this datas and how we see it's given in this summary. By this data.describe we concluded that types of informations. 